pwnlib/util/cyclic.py

```diff
@@ -344,3 +329,128 @@ def _gen_find(subseq, generator):
+    if isinstance(alphabet, six.text_type):
+        return ''.join(seq)
+    elif isinstance(alphabet, bytes):
+        return bytes(bytearray(seq))
+    else:
+        return seq
+
+class cyclic_gen(object):
+    """
+    Creates a stateful cyclic generator which can generate sequential chunks of de Bruijn sequences.
+
+    >>> g = cyclic_gen() # Create a generator
+    >>> g.get(4) # Get a chunk of length 4
+    b'aaaa'
+    >>> g.get(4) # Get a chunk of length 4
+    b'baaa'
+    >>> g.get(8) # Get a chunk of length 8
+    b'caaadaaa'
+    >>> g.get(4) # Get a chunk of length 4
+    b'eaaa'
+    >>> g.find(b'caaa') # Position 8, which is in chunk 2 at index 0
+    (8, 2, 0)
+    >>> g.find(b'aaaa') # Position 0, which is in chunk 0 at index 0
+    (0, 0, 0)
+    >>> g.find(b'baaa') # Position 4, which is in chunk 1 at index 0
+    (4, 1, 0)
+    >>> g.find(b'aaad') # Position 9, which is in chunk 2 at index 1
+    (9, 2, 1)
+    >>> g.find(b'aada') # Position 10, which is in chunk 2 at index 2
+    (10, 2, 2)
+    >>> g.get() # Get the rest of the sequence
+    b'faaagaaahaaaiaaajaaa...yyxzyzxzzyxzzzyyyyzyyzzyzyzzzz
+    >>> g.find(b'racz') # Position 7760, which is in chunk 4 at index 7740
+    (7760, 4, 7740)
+    >>> g.get(12) # Generator is exhausted
+    Traceback (most recent call last):
+    File "<stdin>", line 1, in <module>
+    File ".../pwntools/pwnlib/util/cyclic.py", line 368, in get
+        out = [next(self._generator) for _ in range(length)] if length != None else [next(self._generator)] + list(self._generator)
+    File ".../pwntools/pwnlib/util/cyclic.py", line 368, in <listcomp>
+        out = [next(self._generator) for _ in range(length)] if length != None else [next(self._generator)] + list(self._generator)
+    StopIteration
+
+    >>> g = cyclic_gen(string.ascii_uppercase, n=8) # Custom alphabet and item size
+    >>> g.get(12) # Get a chunk of length 12
+    'AAAAAAAABAAA'
+    >>> g.get(18) # Get a chunk of length 18
+    'AAAACAAAAAAADAAAAA'
+    >>> g.find('CAAAAAAA') # Position 16, which is in chunk 1 at index 4
+    (16, 1, 4)
+    """
+
+    def __init__(self, alphabet = None, n = None):
+        if n is None:
+            n = context.cyclic_size
+
+        if alphabet is None:
+            alphabet = context.cyclic_alphabet
+
+        self._generator = de_bruijn(alphabet, n)
+        self._alphabet = alphabet
+        self._total_length = 0
+        self._n = n
+        self._chunks = []
+
+    def get(self, length = None):
+        """
+        Get the next de Bruijn sequence from this generator.
+
+        >>> g = cyclic_gen()
+        >>> g.get(4) # Get a chunk of length 4
+        b'aaaa'
+        >>> g.get(4) # Get a chunk of length 4
+        b'baaa'
+        >>> g.get(8) # Get a chunk of length 8
+        b'caaadaaa'
+        >>> g.get(4) # Get a chunk of length 4
+        b'eaaa'
+        >>> g.get() # Get the rest of the sequence
+        b'faaagaaahaaaiaaajaaa...yyxzyzxzzyxzzzyyyyzyyzzyzyzzzz
```

Comment:
You missed the final quote
```suggestion
        b'faaagaaahaaaiaaajaaa...yyxzyzxzzyxzzzyyyyzyyzzyzyzzzz'
```

Reply:
Wow, how sloppy of me. Fixed.